src/Resource/GetSpecificRate.php

```diff
@@ -39,12 +39,15 @@ public function fetch(ImportConnector $connector): \Iterator
             true
         );
 
-        $base = $response['asset_id_base'];
-
-        $quote = $response['asset_id_quote'];
-        $rate = $response['rate'];
-        $time = $response['time'];
-
-        return new SpecificRateRecord($time, $base, $quote, $rate);
+        $rate = function () use ($response) {
```

Comment:
Again, this seems to be a very roundabout way of copying an array. Since we don't need the `SpecificRateRecord` collection, this entire code block (lines 42-51) can be replaced with `yield $response`.